test/if.civet

```diff
@@ -859,8 +859,8 @@ describe "if", ->
       else
         b
       ---
-      let ref, y;x = (ref = f())?
-        {y} = ref,a
+      let ref, y;x = ((ref = f()))?
+        ({y} = ref,a)
```

Comment:
Hmm, maybe these are two settings where we should discriminate based on parent or grandparent?
* if expressions already wrap the condition in parens when writing `?:`. Ideally we could detect this situation and not double wrap here...
* I didn't expect the `then` clause to be wrapped. Does eslint want it to be?

Reply:
If the `then` clause isn't wrapped, it's actually not valid JS, because of the precedence of the comma operator relative to the ternary.

Reply:
To answer the first question though, ESLint requires double-wrapping with `?:` as well as `if` and `while`.